WordPress/openverse
# Fix frontend Sentry configuration

## Fixes



Fixes #2406 by @sarayourfriend 

## Description


Some of the changes in this PR are not strictly necessary. Specifically, removing `DISABLE_SENTRY` and upgrading the sentry dependency. However, `DISABLE_SENTRY` doesn't appear to be necessary (it was disabling itself _just fine_ without us even trying to in production (:crying_cat_face:)), Sentry disables itself if no DSN exists and this was only used for Playwright, which runs its own Nuxt build, so there shouldn't be any leakage from the base Dockerfile that has the DSN now. Upgrading the Sentry module also isn't necessary but why not, if we're already in here anyway :shrug:

The actual fixes in this PR:

1. Adding `SENTRY_DSN` as a build-time environment variable in the frontend's Dockerfile. This is the most important one as it is what actually configures Sentry to be enabled. The Nuxt Sentry module depends on the fact that it can configure Sentry at build time. There is no option to leave this to a run-time configuration, not if we also want client reporting, unless we did some special shenanigans to inject the environment variable at SSR time. Using public runtime config does _not_ solve this for the _client_, only the server.
2. Adding `config.environment` to the public runtime config for Sentry. This aspect of Sentry's configuration is _not_ bypassed at build time if SENTRY_DSN is not available to the module, but it won't automatically pull `SENTRY_ENVIRONMENT` from the runtime environment _for the client side_, so we need to manually pass that here (so that the client-side Sentry knows if it is staging or production). We add it to `config` because annoyingly, the Sentry module hard-codes `environment` to `NODE_ENV` at build time, meaning it'll always be hard-coded to `production`, even in the staging runtime, and the Sentry SDK will skip checking `SENTRY_ENVIRONMENT` if `environment` is explicitly passed to Sentry's `init` function (which the Nuxt module handles for us).

Basically: Nuxt module hidden complexity and assumptions bite us again :confused:

## Testing Instructions

Confirm that the fix works in theory by running `SENTRY_DSN=<valid-dsn> pnpm exec nuxt build` in the `frontend` directory followed by `SENTRY_ENVIRONMENT=local pnpm exec nuxt start`. You'll get CORS errors because Sentry isn't configured to accept requests from localhost, so it's probably just easiest for us to check this in staging by manually causing an error on the frontend staging after deployment.

## Checklist


- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
